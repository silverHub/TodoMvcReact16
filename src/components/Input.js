import './Input.css';
import React from 'react';
import PropTypes from 'prop-types';

class Input extends React.Component {
	state = {
		value: ''
	};

	onKeyDown = event => {
		if (event.key === 'Enter') {
			this.props.onEnter(this.state.value);
			this.setState({ value: ''});
		}
	};

	render() {
		/* Controlled native input*/
		return <input className="new-todo"
					  placeholder="What needs to be done?"
					  value={this.state.value}
					  onChange={event => this.setState({value: event.target.value})}
					  onKeyDown={this.onKeyDown}
					  autoFocus/>;
	}

}

Input.propTypes = {
	onEnter: PropTypes.func
};

export default Input;
