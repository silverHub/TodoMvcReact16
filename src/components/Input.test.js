import React from 'react';
import Input from './Input';
import { shallow } from 'enzyme';


describe('testing Input component', () => {

	test('if renders', () => {
		expect(shallow(<Input />)).toMatchSnapshot();
	});

	test('if returns typed text on onEnter', () => {
		const mockFn = jest.fn();
		const wrap = shallow(<Input onEnter={mockFn} />);

		// type something
		 wrap.find('input').simulate('change', { target: { value: 'typedInText' }});

		// hit Enter
		wrap.find('input').simulate('keyDown', { key: 'Enter'});
		expect(mockFn).toBeCalledWith('typedInText');
	});
});

