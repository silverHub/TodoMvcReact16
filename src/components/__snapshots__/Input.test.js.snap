// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Input if renders 1`] = `
<input
  autoFocus={true}
  className="new-todo"
  onChange={[Function]}
  onKeyDown={[Function]}
  placeholder="What needs to be done?"
  value=""
/>
`;
